### PR TITLE
improve globbing (fixes #91)

### LINKED ISSUE #91: allow ** globbing operator at beginning of relative path
`**/<locale_code>.yml` should work

### PR DESCRIPTION
**Pull request by alexanderwilling**
Partly addresses #92

### CODE REVIEW

File: internal/paths/glob_test.go [hunk 1]
```diff
@@ -83,6 +137,7 @@ func TestGlob(t *testing.T) {
 }
 
 func areEqual(s, t []string) bool {
+	log.Println("got", s, "want", t)
```
Comment (sacry-dyn) [inline internal/paths/glob_test.go]: @alexanderwilling (Question) sure we want logging enabled here?

File: internal/paths/glob.go [hunk 2]
```diff
@@ -4,22 +4,33 @@ import (
...
 // Glob supports * and ** globbing according to https://phraseapp.com/docs/developers/cli/configuration/#globbing
 func Glob(pattern string) (matches []string, err error) {
 	pattern = filepath.Clean(pattern)
+	pattern = escape(pattern)
 
 	if strings.Count(pattern, dirGlobOperator) > 1 {
 		return nil, fmt.Errorf("invalid pattern '%s': the ** globbing operator may only be used once in a pattern", pattern)
 	}
 
+	if dirGlobOperatorUseValid(pattern) {
```
Comment (sacry-dyn) [inline internal/paths/glob.go]: @alexanderwilling The construct is positive, why is it true and we fail?

File: internal/paths/glob.go [hunk 3]
```diff
@@ -4,22 +4,33 @@ import (
...
+	"runtime"
 	"strings"
 )
 
-var dirGlobOperator = string(filepath.Separator) + "**" + string(filepath.Separator)
+var dirGlobOperator = "**"
+
+// dirGlobOperatorUseValid returns false if '**' occurs, but '/**/' doesn't and pattern does not start with '**/'.
+func dirGlobOperatorUseValid(pattern string) bool {
+	return strings.Contains(pattern, "**") && (!strings.Contains(pattern, string(filepath.Separator)+"**"+string(filepath.Separator)) && !strings.HasPrefix(pattern, "**"+string(filepath.Separator)))
```
Comment (sacry-dyn) [inline internal/paths/glob.go]: @alexanderwilling (Improvement) Would split this up and state it in a "positive" way.

File: internal/paths/glob_test.go (hunk 1 above)
Comment (alexanderwilling) [inline internal/paths/glob_test.go]: Oops, that's a leftover. Thanks for pointing it out, I'll remove it.

File: internal/paths/glob.go (hunk 2 above)
Comment (alexanderwilling) [inline internal/paths/glob.go]: Yeah, the '!' got lost in renaming it like 5 times... Thanks

**Comment by alexanderwilling** [comment]
integrations test pass, too: http://awsbuild.phraseapp.io/job/phraseapp-integration/723/

**Comment by sacry-dyn** [comment]
@alexanderwilling lgtm

**Pull request merged**